aws-logs-resourcepolicy/src/main/java/software/amazon/logs/resourcepolicy/BaseHandlerStd.java

```diff
@@ -0,0 +1,42 @@
+package software.amazon.logs.resourcepolicy;
+
+import software.amazon.awssdk.awscore.exception.AwsServiceException;
+import software.amazon.awssdk.services.cloudwatchlogs.CloudWatchLogsClient;
+import software.amazon.awssdk.services.cloudwatchlogs.model.DescribeMetricFiltersRequest;
+import software.amazon.awssdk.services.cloudwatchlogs.model.DescribeMetricFiltersResponse;
+import software.amazon.awssdk.services.cloudwatchlogs.model.DescribeResourcePoliciesResponse;
+import software.amazon.cloudformation.exceptions.BaseHandlerException;
+import software.amazon.cloudformation.exceptions.CfnNotFoundException;
+import software.amazon.cloudformation.proxy.*;
+
+public abstract class BaseHandlerStd extends BaseHandler<CallbackContext> {
+  @Override
+  public final ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+    final AmazonWebServicesClientProxy proxy,
+    final ResourceHandlerRequest<ResourceModel> request,
+    final CallbackContext callbackContext,
+    final Logger logger) {
+    return handleRequest(
+      proxy,
+      request,
+      callbackContext != null ? callbackContext : new CallbackContext(),
+      proxy.newProxy(ClientBuilder::getLogsClient),
+      logger
+    );
+  }
+
+  protected abstract ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+    final AmazonWebServicesClientProxy proxy,
+    final ResourceHandlerRequest<ResourceModel> request,
+    final CallbackContext callbackContext,
+    final ProxyClient<CloudWatchLogsClient> proxyClient,
+    final Logger logger);
+
+  protected boolean policyExists(CloudWatchLogsClient client, ResourceModel model) {
+    DescribeResourcePoliciesResponse response = client.describeResourcePolicies();
```

Comment:
Credentials are injected by this `proxy.newProxy(ClientBuilder::getLogsClient)`.

I'm not sure what this means `doesn't like like we passed through a request here either.`.  `client.describeResourcePolicies()` builds it's own empty request.

Reply:
only issue with leaving request empty is that we should iterate through all resource policies until nextToken is null in the case that the resource policy we are looking for is not in the first batch. Can we add a do, while loop here to avoid missing anything?